src/store/matchedAudiences.ts

```diff
@@ -30,23 +39,74 @@ class MatchedAudienceStore {
     this._save();
   }
 
-  _save() {
+  _save(): void {
     storage.set(StorageKeys.MATCHED_AUDIENCES, this.matchedAudiences);
     storage.set(StorageKeys.MATCHED_AUDIENCE_IDS, this.matchedAudienceIds);
   }
 
-  setMatchedAudiences(newlyMatchedAudiences: MatchedAudience[]) {
-    // TODO: decide if we need to check duplicate audiences here...
+  setMatchedAudiences(matchedAudiences: MatchedAudience[]): void {
+    this.matchedAudiences = matchedAudiences;
+    this.matchedAudienceIds = matchedAudiences.map((audience) => audience.id);
+  }
 
-    if (newlyMatchedAudiences.length === 0) return;
-    const allMatchedAudiences = [
-      ...this.matchedAudiences,
-      ...newlyMatchedAudiences,
-    ];
-    this.matchedAudiences = allMatchedAudiences;
-    this.matchedAudienceIds = allMatchedAudiences.map(
-      (audience) => audience.id
+  getMatchedAudiences(): MatchedAudience[] {
+    return [...this.matchedAudiences];
+  }
+
+  saveMatchingAudiences(
+    audienceDefinitions: AudienceDefinition[],
+    viewStore: ViewStore
+  ): void {
```

Comment:
@ydennisy 
I've committed the changes as a WIP so you can take a look while I'm moving.

There is a coupling between `viewStore` and `matchedAudienceStore` in the `saveMatchingAudiences` method.
I'll think in a better way when I'm back online. =)